src/rust/ide/src/controller/text.rs

```diff
@@ -46,32 +46,36 @@ enum FileHandle {
 /// This struct contains all information and handles to do all module controller operations.
 #[derive(Clone,CloneRef,Debug)]
 pub struct Handle {
+    logger: Logger,
     file: FileHandle,
 }
 
 impl Handle {
-
-    /// Create controller managing plain text file (which is not a module).
-    pub fn new_for_plain_text
-    (path:FilePath, language_server:Rc<language_server::Connection>) -> Self {
-        let path = Rc::new(path);
-        Self {
-            file : FileHandle::PlainText {path,language_server}
-        }
-    }
-
-    /// Create controller managing Luna module file.
-    pub fn new_for_module(controller:controller::Module) -> Self {
-        Self {
-            file : FileHandle::Module {controller}
-        }
+    /// Create a Text Controller for file.
+    ///
+    /// This constructor checks what kind of file we read, and load it as a module file or plain
+    /// text file.
+    pub async fn new
```

Comment:
So we won't have any API to open a module file in a plain text mode?
(not saying this is bad thing)

Reply:
If we need to, we can add then such method. But actually I think we always want to open modules as modules, to avoid discrepancies between graph views and opened files.